scripts/choose_asvs.py

```diff
@@ -0,0 +1,132 @@
+"""
+Try to identify which asvs can be ignored for a PR.
+"""
+import os
+import shlex
+import subprocess
+
+here = os.path.dirname(__file__)  # assumes repo directory
+bmark_dir = os.path.join("asv_bench", "benchmarks")
+
+
+def find_changed():
+    """
+    Find the names of files changes (compared to master) in this branch.
+    """
+    cmd = "git diff upstream/master --name-only"
+
+    p = subprocess.Popen(
+        shlex.split(cmd),
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        close_fds=True,
+    )
+    stdout, stderr = p.communicate()
+
+    flist = stdout.decode("utf-8").splitlines()
+    return flist
+
+
+def find_asv_paths():
+    """
+    Find all of our asv benchmark files.
+    """
+    bmarks = os.path.join(here, bmark_dir)
+    walk = os.scandir(bmarks)
+    walk = [
+        x
+        for x in walk
+        if x.is_file()
+        and x.name.endswith(".py")
+        and x.name not in ["__init__.py", "pandas_vb_common.py"]
+    ]
+
+    paths = [x.path.replace(bmarks, "").lstrip(os.sep) for x in walk]
+    return paths
+
+
+def trim_asv_paths():
+    """
+    Exclude benchmark files we can be confident are not affected
+    by this PR.
+    """
+    flist = find_changed()
+
+    changed = [x for x in flist if "pandas/" in x and "pandas/tests/" not in x]
+
+    asv_paths = find_asv_paths()
+
+    if "setup.py" in flist:
+        # We can't exclude anything
+        return asv_paths
+    if any("_libs/src" in x for x in changed):
```

Comment:
We do have dependencies for these mostly declared in the `setup.py` file. In the future could potentially use that to refine this further